src/js/components/generateFiles/GenerateFilesContent.jsx

```diff
@@ -95,50 +122,73 @@ export default class GenerateFilesContent extends React.Component {
                         </div>
                     </div>
 
-                    <div className="agency-toggle">
-                        <div className="agency-toggle__text">
-                            Generate File D1 and D2 from records where my agency is the:
-                        </div>
-                        <AgencyToggle
-                            funding={this.props.fundingAgency}
-                            toggleAgencyType={this.props.toggleAgencyType} />
-                        <span className="agency-toggle__info-icon-holder">
-                            <div ref={(div) => {
-                                this.referenceDiv = div;
-                            }}>
-                                {tooltip}
-                                <button
-                                    id="agency-toggle__info-icon"
-                                    className="agency-toggle__info-icon"
-                                    onFocus={this.showTooltip}
-                                    onBlur={this.closeTooltip}
-                                    onMouseLeave={this.closeTooltip}
-                                    onMouseEnter={this.showTooltip} >
-                                    <InfoCircle />
-                                </button>
-                            </div>
-                        </span>
-                    </div>
-
                     <div className="usa-da-generate-content">
+                        <div className="agency-toggle">
+                            <div className="agency-toggle__text">
+                                Generate File D1 from records where my agency is the:
+                            </div>
+                            <AgencyToggle
+                                funding={d1.fundingAgency}
+                                toggleAgencyType={this.toggleAgencyType.bind(this, "d1")} />
+                            <span className="agency-toggle__info-icon-holder">
+                                <div ref={(div) => {
+                                    this.d1ReferenceDiv = div;
+                                }}>
+                                    {d1Tooltip}
+                                    <button
+                                        id="agency-toggle__info-icon"
+                                        className="agency-toggle__info-icon"
+                                        onFocus={this.showTooltip.bind(this, 'd1')}
+                                        onBlur={this.closeTooltip.bind(this, 'd1')}
+                                        onMouseLeave={this.closeTooltip.bind(this, 'd1')}
+                                        onMouseEnter={this.showTooltip.bind(this, 'd1')} >
+                                        <InfoCircle />
+                                    </button>
+                                </div>
+                            </span>
+                        </div>
                         <GenerateFileBox
```

Comment:
A lot of duplicate code here, perhaps could be abstracted into class method.

Reply:
Yeah, let me take another look at this.